flink-kubernetes-operator/src/main/java/org/apache/flink/kubernetes/operator/observer/Observer.java

```diff
@@ -75,12 +80,15 @@ private boolean isApplicationClusterReady(FlinkDeployment dep) {
     }
 
     private void observeJmDeployment(
-            FlinkDeployment flinkApp, Context context, Configuration effectiveConfig) {
+            FlinkDeployment flinkApp,
+            Context context,
+            Configuration effectiveConfig,
+            boolean checkOnReady) {
         FlinkDeploymentStatus deploymentStatus = flinkApp.getStatus();
         JobManagerDeploymentStatus previousJmStatus =
                 deploymentStatus.getJobManagerDeploymentStatus();
 
-        if (JobManagerDeploymentStatus.READY == previousJmStatus) {
+        if (JobManagerDeploymentStatus.READY == previousJmStatus && !checkOnReady) {
```

Comment:
Rather than adding the flag, this can be moved into observe.

Reply:
@tweise +1, the branch that the REST server to be ready is only covered in `observe` method. The exception like `TimeoutException` doesn't need to call `isJobManagerPortReady` to check.